vernac/declare.ml

```diff
@@ -734,14 +744,20 @@ let mutual_make_bodies ~typing_flags ~fixitems ~rec_declaration ~possible_indexe
     let vars = Vars.universes_of_constr (List.hd fixdecls) in
     vars, fixdecls, None
 
-let declare_mutually_recursive ~info ~cinfo ~opaque ~uctx ~rec_declaration ~possible_indexes =
+let declare_mutually_recursive ~info ~cinfo ~opaque ~uctx ~rec_declaration ~possible_indexes ?using () =
   let { Info.poly; udecl; scope; clearbody; kind; typing_flags; user_warns; ntns; _ } = info in
+  let env = Global.env() in
   let vars, fixdecls, indexes =
-    mutual_make_bodies ~typing_flags ~fixitems:cinfo ~rec_declaration ~possible_indexes in
+    mutual_make_bodies env ~typing_flags ~fixitems:cinfo ~rec_declaration ~possible_indexes in
   let uctx = UState.restrict uctx vars in
   let univs = UState.check_univ_decl ~poly uctx udecl in
+  let evd = Evd.from_env env in
+  (* let using = Option.map (interp_proof_using_gen f env evd [name, typ, body]) using in *)
```

Comment:
@herbelin that would the point to collect the mutual info data and call interp_proof_using only once with the data, if we would like to. I've left this code here as a placeholder; maybe I should add a comment instead.

Reply:
Done in #18998

Reply:
Yes, I would call interp_proof_using only once with a cinfo argument so as to get all `fixnames` (but maybe with a comment that the body is already a mutual `Fix(...)` that contains all the data).

But that's also dependent on the meaning we want to give to `using="Type"`, that is whether it should include the body or not (see #18860).

Reply:
Yes, I moved it to a different PR.

Reply:
OK